libsolidity/analysis/ConstantEvaluator.h

```diff
@@ -43,33 +43,42 @@ class TypeChecker;
 class ConstantEvaluator: private ASTConstVisitor
 {
 public:
-	ConstantEvaluator(
-		langutil::ErrorReporter& _errorReporter,
-		size_t _newDepth = 0,
-		std::shared_ptr<std::map<ASTNode const*, TypePointer>> _types = std::make_shared<std::map<ASTNode const*, TypePointer>>()
-	):
-		m_errorReporter(_errorReporter),
-		m_depth(_newDepth),
-		m_types(std::move(_types))
+	struct TypedRational
 	{
-	}
+		// TODO for RationalNumberType the value is redundant.
```

Comment:
Maybe make `value` `optional` to express that?

Reply:
Acutally value is the one that will be used...